hicommonwealth/commonwealth
# Compound proposal proxy

## Link to Issue
Closes: #4773 

## Description of Changes
- Adds support for fetching alpha, bravo and compatibility-bravo governance proposals and votes via `/proposals` and `/votes` respectively
- Use Redis to cache Compound governance contract version (alpha vs bravo vs compatibility-bravo)
- Removed unused code in Compound's api, chain, and governance controllers.
  - Removed chain-entity based code in the governance controller
- Used React query to fetch Compound proposals

## Test Plan
- Load `/tribe/proposals`
- Load a specific tribe proposal and ensure votes also load (ensure going back to proposal page works via the back button)
- Copy a specific proposals page and open it directly in a new tab (i.e. open a proposal without navigating through the proposals page first)
- View osmosis and aave proposal pages to ensure they work as usual.

## Deployment Plan

1. 

## Other Considerations

- MUST NOT be merged before #4697 since this PR is dependent upon those changes.
- NOTE: this PR adds support for Compound proposal loading on Ethereum _only_. Compound proposals from Celo and Fantom cannot be loaded with this strategy due to slow public nodes, rate-limiting, or slightly altered contract implementations (i.e. performance/implementation discrepancies across chains). For example, moola-market has an incorrect `initialProposalId` set to 0 when in reality the correct id is 1. Fetching proposal 0 throws an error and fetching all proposal created events takes several minutes (rate limited on subsequent retries). Support for these chains needs to have a different approach which either includes EVM 2.0 or an indexer like TheGraph or SubQuery. Either way, generalized fetching of Compound-like proposals from any EVM chain isn't feasible. **Contracts should/need a personalized approach in order to eliminate all bugs and have the best performance.**
